### DISCUSSION TITLE
If you are a litigator, what do you think of this idea...

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#4130] [General]
Suppose you receive a brief with a memorandum from the other side, similar to the attached. The brief cites many cases with rules and arguments they support. What happens next? Do you ask a paralegal or junior lawyer to:

1. Confirm that the citations are correct?
2. Find any errors to undermine the other side's stance?
3. Verify that those cases indeed support the rules the other side claims they do?

I wanted to see if any automation of this process is feasible. Here is a [prototype](http://108.181.215.210:8501/) ("Analyze legal brief" tab). The goal is to extract all cited cases, pull their decisions from a database, read the cases, and confirm point #3. The analysis may take 20-30 seconds to complete in some cases. Please be patient. 

What do you think? Is this useful, or is there already an existing solution for this workflow? What other features could be useful? 

Cheers, 

PS. The tool at the moment relies on CL database. It may not find all the cited cases yet. 
[Brief - Appellant.pdf](https://github.com/user-attachments/files/15937517/Brief.-.Appellant.pdf)

**Comment by mlissner** [comment]
Awe, no fun: 

```
RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}
```

But the idea is sound. I think this is a lot like Cara, no?

**Reply by legaltextai** [reply]
Sorry about that. I never quite get when OAI rate limits are going to hit. I will switch to another LLM shortly. 
I wasn't aware of Cara, just looked at it (the one from casetext/lexis, right?) Their functionality is much broader. I am focusing on just one minor workflow: extract all cited cases with rules those cases allegedly support - confirm whether they are correct or not.

**Reply by mlissner** [reply]
Yeah, this is all really neat. The check rule thing broke on me though:

```
File "/home/administrator/.local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 600, in _run_script
    exec(code, module.__dict__)
File "/home/administrator/helpmefindcase/case_summary/main.py", line 137, in <module>
    result = check_case_and_rule(case_name, rule, i)
File "/home/administrator/helpmefindcase/case_summary/main.py", line 107, in check_case_and_rule
    opinion = get_citation_and_html(case_name)
File "/home/administrator/.local/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 165, in wrapper
    return cached_func(*args, **kwargs)
File "/home/administrator/.local/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 194, in __call__
    return self._get_or_create_cached_value(args, kwargs)
File "/home/administrator/.local/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 221, in _get_or_create_cached_value
    return self._handle_cache_miss(cache, value_key, func_args, func_kwargs)
File "/home/administrator/.local/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 277, in _handle_cache_miss
    computed_value = self._info.func(*func_args, **func_kwargs)
File "/home/administrator/helpmefindcase/case_summary/modules/brief_analysis.py", line 34, in get_citation_and_html
    sub_opinions_url = citation_data[0]['clusters'][0]['sub_opinions'][0]
```

Have you seen our citation look up tool? One of our board members has been saying for a while it should have a PDF upload functionality: https://www.courtlistener.com/c/

How do you identify the rule, and how do you check it?

**Reply by legaltextai** [reply]
it's probably because there was no corresponding case (looked up by citation) in the CL. i am getting a lot of those b/c the indiana cases are a bit out of date in CL. if it finds the cited cases, the outcome should look smth like the attached. 
<img width="787" alt="Screenshot 2024-06-24 at 11 59 58 AM" src="https://github.com/freelawproject/courtlistener/assets/144342123/17c7fe6f-8031-4823-95ad-16ad81004780">
if you give me the case that you clicked on , i could look into more detail. 
the logic here is as follows: 
- the model extracts rule / corresponding case from a brief in json format; 
- the code reiterates over each case in the output: 
-- extracts citation by using CL citation lookup API 
-- extracts the text of the opinion from 'sub-opinion' URL 
- the rule and the full text of the opinion is fed into another model asking to review and extract most relevant quotes from the opinion related to the rule. 
there is still much room for improvement and testing. this is just an early prototype to test the idea.

**Reply by mlissner** [reply]
> this is just an early prototype to test the idea.

Pretty cool though for a quick prototype, I'll grant you that! 

I don't think I have much feedback to give, but I think it's a great idea and something that people would take advantage of, if they knew it existed.

**Comment by auc0le** [comment]
This looks like a cool idea, what ever happened to the project?

**Reply by mlissner** [reply]
Just a prototype so far, unless @legaltextai has more to say. :)
